File: code/back-office/server/src/main/java/internship/project/election/domain/Role.java
package internship.project.election.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "roles")
public class Role extends AbstractEntity<Integer> {

    @Column(name = "nom")
    private String name;

    @Column(name = "niveau")
    private Integer level;
}

File: code/back-office/front/src/app/dto/role.ts
import { Privileges } from '../privileges';

export class Role {
  id: number;
  name: string;
  level: number;

  constructor(id: number, name: string, level: number) {
    this.id = id;
    this.name = name;
    this.level = level;
  }

  getName(): string {
    return this.name;
  }
}
